pkg/apis/networking/v1alpha1/ingress_types.go

```diff
@@ -121,6 +128,13 @@ const (
 	// IngressVisibilityClusterLocal is used to denote that the Ingress
 	// should be only be exposed locally to the cluster.
 	IngressVisibilityClusterLocal IngressVisibility = "ClusterLocal"
+	// The knative ingress will be able to serve HTTP connections.
+	HTTPOptionEnabled HTTPOption = "Enabled"
+	// The knative ingress will reject HTTP traffic
+	HTTPOptionDisabled HTTPOption = "Disabled"
```

Comment:
Per the discussion in the [doc](https://docs.google.com/document/d/1yVOGo-j1hI7oCpuWVIi5iBJwX1wKbOzWaV0QNJxCbaY/edit?disco=AAAAKYgMWoc), we will not support Disabled (at least in the beginning) as it may break the HTTP01 challenge contract of managed TLS feature.